phoenix-core-server/src/main/java/org/apache/phoenix/coprocessor/PhoenixMasterObserver.java

```diff
@@ -0,0 +1,187 @@
+     * @param c           the environment to interact with the framework and master
+     * @param regionInfoA the left daughter region
+     * @param regionInfoB the right daughter region
+     */
+    @Override
+    public void postCompletedSplitRegionAction(final ObserverContext<MasterCoprocessorEnvironment> c,
+                                               final RegionInfo regionInfoA, final RegionInfo regionInfoB) {
+        Configuration conf = c.getEnvironment().getConfiguration();
+        try {
+            Connection conn  = QueryUtil.getConnectionOnServer(conf);
+            // CDC will be enabled on Phoenix tables only
+            PTable phoenixTable = getPhoenixTable(conn, regionInfoA.getTable());
+            if (phoenixTable == null) {
+                LOGGER.info("Not a Phoenix Table, skipping partition metadata update.");
+                return;
+            }
+            // find streamName with ENABLED status
+            String tableName = phoenixTable.getName().getString();
+            PreparedStatement pstmt = conn.prepareStatement(STREAM_STATUS_QUERY);
+            pstmt.setString(1, tableName);
+            ResultSet rs = pstmt.executeQuery();
+            if (rs.next()) {
+                String streamName = rs.getString(1);
+                String parentPartitionID = getParentPartitionId(conn, tableName, streamName, regionInfoA, regionInfoB);
+                upsertDaughterPartition(conn, tableName, streamName, parentPartitionID, regionInfoA);
+                upsertDaughterPartition(conn, tableName, streamName, parentPartitionID, regionInfoB);
+                updateParentPartitionEndTime(conn, tableName, streamName, parentPartitionID, regionInfoA.getRegionId());
+            }
+        } catch (SQLException e) {
+            LOGGER.error("Unable to update CDC Stream Partition metadata: " + e);
+        }
+    }
+
+    private PTable getPhoenixTable(Connection conn, TableName tableName) throws SQLException {
+        PTable pTable;
+        try {
+            pTable = PhoenixRuntime.getTable(conn, tableName.toString());
+        } catch (TableNotFoundException e) {
+            return null;
+        }
+        return pTable;
+    }
+
+    /**
+     * Lookup parent's partition id (region's encoded name) in SYSTEM.CDC_STREAM.
+     * We infer start and end key for the parent from the two daughters' start/end keys.
+     */
+    private String getParentPartitionId(Connection conn, String tableName, String streamName,
+                                        RegionInfo regionInfoA, RegionInfo regionInfoB)
+            throws SQLException {
+        byte[] inferredParentStartKey =
+                Bytes.compareTo(regionInfoA.getStartKey(), regionInfoB.getStartKey()) < 0 ?
+                regionInfoA.getStartKey() : regionInfoB.getStartKey();
+
+        byte[] inferredParentEndKey = (regionInfoB.getEndKey().length == 0 || regionInfoA.getEndKey().length == 0)
```

Comment:
`regionInfoA` endkey should not be empty? or is it possible that regionInfoA and regionInfoB could be swapped?